src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,73 @@
+package duke.command;
+
+import duke.Storage;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.ui.Ui;
+
+import static duke.ui.Message.MESSAGE_ADD;
+import static duke.ui.Message.concatLines;
+import static duke.ui.Message.getTaskTotalMsg;
+
+/**
+ * Adds a task to the task list.
+ */
+public class AddCommand extends Command {
+    private String taskType;
+    private String taskParams;
+    private Task newTask;
+
+    /**
+     * Constructs an AddCommand object given task type and task parameters.
+     *
+     * @throws DukeException if any of the arguments are invalid
+     */
+    public AddCommand(String taskType, String taskParams) {
+        this.taskType = taskType;
+        this.taskParams = taskParams;
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
Nice! You abstracted the commands to add a task, event and deadline into a single `AddCommand` class.